paddle/trainer/NewRemoteParameterUpdater.cpp

```diff
@@ -66,28 +69,68 @@ void NewRemoteParameterUpdater::init(
   // from parameter server
   if (paddle_begin_init_params(parameterClient_)) {
     LOG(INFO) << "paddle_begin_init_params start";
+    if (!optimizerConfigNew_.has_optimizer()) {
+      // NOTE: convert V1 OptimizatioinConfig proto to OptimizerConfig if
+      // OptimizerConfig not set. This makes golang pserver compatible with
+      // handy V1 demos. Support SGD only.
+      // TODO: Remove these lines when golang pserver is production ready
+      for (int i = 0; i < parameterSize(); ++i) {
+        auto paramConfig = parameters_[i]->getConfig();
+        LOG(INFO) << "old param config: " << paramConfig.DebugString();
+        auto sgdConfigV2 = optimizerConfigNew_.mutable_sgd();
+        sgdConfigV2->set_momentum(paramConfig.momentum());
+        sgdConfigV2->set_decay(paramConfig.decay_rate());
+        optimizerConfigNew_.set_lr_policy(paddle::OptimizerConfig::Const);
+        auto constlr = optimizerConfigNew_.mutable_const_lr();
+        if (paramConfig.has_learning_rate()) {
+          constlr->set_learning_rate(paramConfig.learning_rate());
+        } else {
+          constlr->set_learning_rate(trainerConfig_.learning_rate());
+        }
+        if (trainerConfig_.algorithm() == "sgd") {
+          optimizerConfigNew_.set_optimizer(paddle::OptimizerConfig::SGD);
+        } else {
+          optimizerConfigNew_.set_optimizer(paddle::OptimizerConfig::SGD);
+        }
+      }
+    }
     for (int i = 0; i < parameterSize(); ++i) {
+      OptimizerConfig optConfigPerParameter;
```

Comment:
It's hard to maintain two version of the same thing. We added config v2 because v1 is a giant proto file which contains everything, which is too hard to understand and blocks the optimizer implementation.
Sorry that you have to write this...